lib/Doctrine/DBAL/SQLParserUtils.php

```diff
@@ -250,11 +251,12 @@ private static function getUnquotedStatementFragments($statement)
             self::ESCAPED_DOUBLE_QUOTED_TEXT . '|' .
             self::ESCAPED_BACKTICK_QUOTED_TEXT . '|' .
             self::ESCAPED_BRACKET_QUOTED_TEXT;
-        $expression = sprintf('/((.+(?i:ARRAY)\\[.+\\])|([^\'"`\\[]+))(?:%s)?/s', $literal);
-
+        $expression = sprintf('/(?:[\s]*--.*?\n)|((.+(?i:ARRAY)\\[.+\\])|([^-\'"`\\[]+))(?:%s)?/s', $literal);
```

Comment:
There's an issue (https://github.com/doctrine/dbal/issues/3472) caused by making this regular expression more complex (https://github.com/doctrine/dbal/pull/3277). I don't know of a proper solution but it seems like making it even more complex may make things even worse.